ibis/sql/postgres/tests/test_udf.py

```diff
@@ -0,0 +1,173 @@
+"""Test support for already-defined UDFs in Postgres"""
+
+import random
+import string
+
+import pytest
+
+import pandas as pd
+import ibis.expr.datatypes
+import ibis.sql.postgres.udf.api
+from ibis.sql.postgres.client import PostgreSQLClient
+
+
+datatypes = ibis.expr.datatypes
+
+# mark test module as postgresql (for ability to easily exclude,
+# e.g. in conda build tests)
+# (Temporarily adding `postgis` marker so Azure Windows pipeline will exclude
+#     pl/python tests.
+#     TODO: update Windows pipeline to exclude postgres_extensions
+#     TODO: remove postgis marker below once Windows pipeline updated
+pytestmark = [
+    pytest.mark.postgresql,
+    pytest.mark.postgis,
+    pytest.mark.postgres_extensions,
+]
+
+# Database setup (tables and UDFs)
+
+
+test_schema = None
+
+
+def gen_schema_name(basename='test'):
+    """Generate a random alpha string starting with 'test_' to be used as a
+    test schema name"""
+    schema_name = '{}_{}'.format(
+        basename,
+        ''.join([random.choice(string.ascii_lowercase) for i in range(6)])
+    )
+    return schema_name
+
+
+def create_test_schema(connection: PostgreSQLClient, max_tries=20):
+    """Create a random test schema, necessary to allow for parallel pytest
+    testing"""
+    schema_exists = True
+    tries = 0
+    while schema_exists and tries < max_tries:
```

Comment:
I just wanted to take into account the possibility that multiple parallel test instances may have created the same random string. It is unlikely (1 in 26**6) but possible. Max tries is probably not necessary, but I was nervous about creating a `while` loop without an explicit cap in case something isn't working right.

Reply:
And as I mentioned in a separate comment, when running tests in parallel mode there can be contention for the same name, and I wasn't sure of a way to ensure a unique sequence ID across processes.

Reply:
Just pushed changes with the session-based scope that took out the multiple attempts at creating a unique schema. Commit [e1ee3bbe60](https://github.com/scottcode/ibis/commit/e1ee3bbe601d3d715f6c6a352543c4594e2fa4a1)